### PR TITLE
Fix DOI/ID message when deaccession a Dataset

### LINKED ISSUE #10309: Correct success message of deaccessioning API endpoint when dataset PID is used
**When does this issue occur?**
When using the API endpoint for deaccessioning a dataset, documented at https://guides.dataverse.org/en/6.1/api/native-api.html#deaccession-dataset, while passing in a dataset's PID instead of its database ID

**What happens?**
The success message reads:
`{'status': 'OK', 'data': {'message': 'Dataset :persistentId deaccessioned for version 1.0'}}`

**To whom does it occur (all users, curators, superusers)?**
All users

**What did you expect to happen?**
In place of ":persistentId", the success message would include either the PID of the dataset or its database ID.

**Which version of Dataverse are you using?**
  v6.1

**Any related open or closed issues to this bug report?**
  https://github.com/IQSS/dataverse/issues/778, https://github.com/IQSS/dataverse/issues/8132

### PR DESCRIPTION
**Pull request by jp-tosca**
**What this PR does / why we need it**:
It was reported on #10309 that this parameter was not substituted by the value on the success message.

**Which issue(s) this PR closes**:
Closes #10309

**Suggestions on how to test this**:

1. Create a Dataset version
2. Deaccession it with `http://localhost:8080/api/datasets/:persistentId/versions/:versionID/deaccession?key=API_KEY&persistentId=doi:10.5072/FK2/RGFWNO`
3. the new message should show as `"message": "Dataset 10.5072/FK2/RGFWNO deaccessioned for version 9"`
4. Check the database id still works `http://localhost:8080/api/datasets/6/versions/10/deaccession?key=API_KEY`
5. message should show as `"message": "Dataset 6 deaccessioned for version 10"`

**Is there a release notes update needed for this change?**:
No

**Additional documentation**:
No

### CODE REVIEW

File: src/main/java/edu/harvard/iq/dataverse/api/Datasets.java [hunk 1]
```diff
@@ -4791,7 +4791,10 @@ public Response deaccessionDataset(@Context ContainerRequestContext crc, @PathPa
                     }
                 }
                 execCommand(new DeaccessionDatasetVersionCommand(req, datasetVersion, false));
-                return ok("Dataset " + datasetId + " deaccessioned for version " + versionId);
+                
+                return ok("Dataset " + 
+                        (":persistentId".equals(datasetId) ? datasetVersion.getDataset().getAuthority() + "/" + datasetVersion.getDataset().getIdentifier() : datasetId) +
```
Comment (pdurbin) [inline src/main/java/edu/harvard/iq/dataverse/api/Datasets.java:4796]: Won't this be missing "doi:"? What if it's "hdl:" instead?

Can we please have a test?

**Review by pdurbin — COMMENTED** [review]
I'm pretty sure this won't show the full PID.

**Comment by jp-tosca** [comment]
> I'm pretty sure this won't show the full PID.

It won't, I removed it. Previously I was using the "protocol" would that be the right thing to use to prepend? @pdurbin

**Comment by jp-tosca** [comment]
Thanks @qqmyers! I updated this PR to use ***asString*** from the GlobalID.

I also added to the tests to check when it is deleted by a specified ID to be contained on the message and a function to delete by global. @pdurbin

File: src/test/java/edu/harvard/iq/dataverse/api/UtilIT.java [hunk 2]
```diff
@@ -3858,6 +3858,19 @@ static Response deaccessionDataset(Integer datasetId, String version, String dea
...
+        if (deaccessionForwardURL != null) {
+            jsonObjectBuilder.add("deaccessionForwardURL", deaccessionForwardURL);
+        }
+        String jsonString = jsonObjectBuilder.build().toString();
+        return given()
+                .header(API_TOKEN_HTTP_HEADER, apiToken)
+                .body(jsonString)
+                .post("/api/datasets/:persistentId/versions/" + version + "/deaccession?persistentId=" + globalId);
+    }
+
```
Comment (pdurbin) [inline src/test/java/edu/harvard/iq/dataverse/api/UtilIT.java:3873]: I guess it's ok to add a whole new method but if you want to pass a PID (I would suggest this term over GlobalID), you could use this pattern you'll see elsewhere in UtilIT:

```
String idInPath = datasetIdOrPersistentId; // Assume it's a number.
String optionalQueryParam = ""; // If idOrPersistentId is a number we'll just put it in the path.
if (!NumberUtils.isCreatable(datasetIdOrPersistentId)) {
    idInPath = ":persistentId";
    optionalQueryParam = "?persistentId=" + datasetIdOrPersistentId;
}
```

**Review by pdurbin — COMMENTED** [review]
This is coming along. I'm adding a suggestion for how to avoid adding an entire new method to UtilIT.

File: src/test/java/edu/harvard/iq/dataverse/api/UtilIT.java (hunk 2 above)
Comment (jp-tosca) [reply src/test/java/edu/harvard/iq/dataverse/api/UtilIT.java:3873]: I made some changes, please let me know if this is better.

Comment (pdurbin) [reply src/test/java/edu/harvard/iq/dataverse/api/UtilIT.java:3873]: It's better, but did you find a datasetIdOrPersistentId example in UtilIT? Do you see what I mean, how we can reduce code duplication?

Comment (jp-tosca) [reply src/test/java/edu/harvard/iq/dataverse/api/UtilIT.java:3873]: I will change that in a bit 😅

Comment (jp-tosca) [reply src/test/java/edu/harvard/iq/dataverse/api/UtilIT.java:3881]: Ok, I made some changes, let me know what you think. 😄

**Review by qqmyers — APPROVED** [review]
Looks good. Tests pass. Looks like all of the prior review comments have been addressed.

**Pull request merged**